src/hooks/usePosts.ts

```diff
@@ -0,0 +1,36 @@
+import { useEffect, useState } from 'react';
+import { Post } from '../types';
+import * as postsService from '../api/posts';
+
+export const usePosts = (selectedUserId: number = 0) => {
```

Comment:
The default value for `selectedUserId` is set to 0. Ensure that this default value is appropriate for your application's logic, as it might lead to unintended behavior if not handled correctly.